nodejs/eks/cni.ts

```diff
@@ -131,7 +134,10 @@ export class VpcCni extends pulumi.dynamic.Resource {
             throw new Error("Could not set VPC CNI options: kubectl is missing. See https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl for installation instructions.");
         }
 
-        const yamlPath = path.join(__dirname, "cni", "aws-k8s-cni.yaml");
+        // Form the relative CNI YAML file path.
+        const projectPath = __dirname.substring(0, __dirname.indexOf("node_modules"));
+        const relativeClassDir = __dirname.replace(projectPath, "");
+        const yamlPath = path.join(relativeClassDir, "cni/aws-k8s-cni.yaml");
```

Comment:
Would it have worked instead to just move `const yamlPath = path.join(__dirname, "cni", "aws-k8s-cni.yaml");` inside each of the functions below?

I'll admit I don't quite follow the three lines above, and hardcoding `node_modules` here feels "wrong" though I'm sure it works okay for normal cases.

Reply:
Moving the `yamlPath` definition into each function did not work. It does not find the file:
`error: Plan apply failed: ENOENT: no such file or directory, open 'cni/aws-k8s-cni.yaml`

> I'll admit I don't quite follow the three lines above, and hardcoding node_modules here feels "wrong"

This was the only way I found to get around embedding an absolute path into the state of the dynamic provider, which seems to be the root issue. Do you have any other suggestions?

Reply:
Fair enough - if this solves the problem then let’s go ahead with it!